File: CS/WindowsApplication3/Main.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraEditors;
using DXSample.Data;

namespace DXSample {
    public partial class Main : XtraForm {
        FilterNameProvider provider;
        public Main() {
            InitializeComponent();
            recordBindingSource.DataSource = DataHelper.GetData(10);
            provider = new FilterNameProvider(gridView1) { AllowSettingFilterNames = true };
        }
        string filePath = "Layout.xml";
        private void OnSaveLayoutButtonClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            gridView1.SaveLayoutToXmlEx(provider, filePath);
        }

        private void OnRestoreLayoutButtonClick(object sender, DevExpress.XtraBars.ItemClickEventArgs e) {
            gridView1.RestoreLayoutFromXmlEx(provider, filePath);
        }
    }
}

File: VB/WindowsApplication3/Main.vb
Imports DevExpress.XtraEditors
Imports DXSample.Data

Namespace DXSample

    Public Partial Class Main
        Inherits XtraForm

        Private provider As FilterNameProvider

        Public Sub New()
            InitializeComponent()
            gridView1.OptionsView.FilterCriteriaDisplayStyle = FilterCriteriaDisplayStyle.Text
            recordBindingSource.DataSource = DataHelper.GetData(10)
            provider = New FilterNameProvider(gridView1) With {.AllowSettingFilterNames = True}
        End Sub

        Private filePath As String = "Layout.xml"

        Private Sub OnSaveLayoutButtonClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs)
            gridView1.SaveLayoutToXmlEx(provider, filePath)
        End Sub

        Private Sub OnRestoreLayoutButtonClick(ByVal sender As Object, ByVal e As DevExpress.XtraBars.ItemClickEventArgs)
            gridView1.RestoreLayoutFromXmlEx(provider, filePath)
        End Sub
    End Class
End Namespace
